python/lsst/daf/butler/core/_butlerUri/file.py

```diff
@@ -217,8 +217,11 @@ def transfer_from(self, src: ButlerURI, transfer: str,
         if transfer not in self.transferModes:
             raise ValueError(f"Transfer mode '{transfer}' not supported by URI scheme {self.scheme}")
 
-        log.debug(f"Transferring {src} [exists: {src.exists()}] -> "
-                  f"{self} [exists: {self.exists()}] (transfer={transfer})")
+        # Existence checks can take time so only try if the log message
```

Comment:
In C++, log macros ensure that log message components are not even evaluated until after it is decided that the message will be issued.  There's no equivalent in Python?  Maybe the log method could take a callable that returns a dict of message components and that would only be called after the threshold check.

Reply:
I've not seen anyone doing deferred calling of log message components in python. We could pass `self.exists` as an argument to some object which then calls it later. This is a bit like how the python docs suggest f-string formatting of log messages is supported: https://docs.python.org/3/howto/logging-cookbook.html#using-custom-message-objects -- can I not worry about this for today and leave it with the isEnabledFor?

Reply:
Of course.  I was just complaining about the state of Python logging, not asking you to fix it :)